src/base_client.py

```diff
@@ -0,0 +1,32 @@
+import hashlib
+import os
+
+
+class BaseShellClient:
+    def __init__(self, host, username, password, port=22):
+        self.host = host
+        self.username = username
+        self.password = password
+        self.port = port
+
+    def calculate_checksum(self, file_path, chunk_size=8192):
```

Comment:
It's utility function, initially you placed this function into utils.py.
Why did you place it in BaseShellClient class ?